plugins/CoreAdminHome/Tasks.php

```diff
@@ -258,6 +262,48 @@ public function updateSpammerBlacklist()
         Option::set(ReferrerSpamFilter::OPTION_STORAGE_NAME, serialize($list));
     }
 
+    public function purgeOrphanedArchives()
+    {
+        $segmentHashesByIdSite = $this->getSegmentHashesByIdSite();
+        $archiveTables = ArchiveTableCreator::getTablesArchivesInstalled('numeric');
+
+        $datesPurged = array();
+        foreach ($archiveTables as $table) {
+            $date = ArchiveTableCreator::getDateFromTableName($table);
+            list($year, $month) = explode('_', $date);
+
+            $dateObj = Date::factory("$year-$month-15");
+
+            $this->archivePurger->purgeDeletedSiteArchives($dateObj);
```

Comment:
just a random thought and not too important... from what I see you pass the date object to generate the archive table name in `getDeletedSiteArchiveIds` and `getDeletedSegmentArchiveIds`. If it makes things simpler, could probably remove a lot of code in line `272-275` and simply directly pass the `tableName`. But I suppose you do it because `ArchivePurger::purgeOutdatedArchives` does it too... makes sense. Probably best to keep it like it is for consistency. (ignore this comment :) )

![image](https://user-images.githubusercontent.com/273120/56180844-2aef5b80-605f-11e9-9575-4d790ba7f24d.png)